uefi/src/proto/network/snp.rs

```diff
@@ -0,0 +1,587 @@
+    pub fn write_nv_data(&self, offset: usize, buffer_size: usize, buffer: *mut c_void) -> Result {
+        (self.nv_data)(self, false, offset, buffer_size, buffer).into()
+    }
+
+    /// Reads the current interrupt status and recycled transmit buffer
+    /// status from a network interface
+    pub fn get_interrupt_status(&self) -> Result<InterruptStatus> {
+        let mut interrupt_status = InterruptStatus::new();
+        let status = (self.get_status)(self, Some(&mut interrupt_status), None);
+        Result::from(status)?;
+        Ok(interrupt_status)
+    }
+
+    /// Reads the current recycled transmit buffer status from a
+    /// network interface
+    pub fn get_recycled_transmit_buffer_status(&self) -> Result<Option<*mut u8>> {
+        let mut tx_buf: *mut c_void = ptr::null_mut();
+        let status = (self.get_status)(self, None, Some(&mut tx_buf));
+        Result::from(status)?;
+        if tx_buf.is_null() {
+            Ok(None)
+        } else {
+            Ok(Some(tx_buf.cast()))
+        }
+    }
+
+    /// Places a packet in the transmit queue of a network interface
+    pub fn transmit(
+        &self,
+        header_size: usize,
+        buffer: &[u8],
+        src_addr: Option<&MacAddress>,
+        dest_addr: Option<&MacAddress>,
+        protocol: Option<&u16>,
+    ) -> Result {
+        (self.transmit)(
+            self,
+            header_size,
+            buffer.len() + header_size,
+            buffer.as_ptr().cast(),
+            src_addr,
+            dest_addr,
+            protocol,
+        )
+        .into()
+    }
+
+    /// Receives a packet from a network interface
+    ///
+    /// On success, returns the size of bytes of the received packet
+    pub fn receive(
+        &self,
+        buffer: &mut [u8],
+        header_size: Option<&mut usize>,
+        src_addr: Option<&mut MacAddress>,
+        dest_addr: Option<&mut MacAddress>,
+        protocol: Option<&mut u16>,
+    ) -> Result<usize> {
+        let mut buffer_size = buffer.len();
+        let status = (self.receive)(
+            self,
+            header_size,
+            &mut buffer_size,
+            buffer.as_mut_ptr().cast(),
+            src_addr,
+            dest_addr,
+            protocol,
+        );
+        Result::from(status)?;
+        Ok(buffer_size)
+    }
+
+    /// Returns a reference to the Simple Network mode
+    #[must_use]
+    pub fn mode(&self) -> &NetworkMode {
+        unsafe { &*self.mode }
+    }
+}
+
+/// Receive unicast packets.
+pub const EFI_SIMPLE_NETWORK_RECEIVE_UNICAST: u32 = 0x01;
+/// Receive multicast packets.
+pub const EFI_SIMPLE_NETWORK_RECEIVE_MULTICAST: u32 = 0x02;
+/// Receive broadcast packets.
+pub const EFI_SIMPLE_NETWORK_RECEIVE_BROADCAST: u32 = 0x04;
+/// Receive packets in promiscuous mode.
+pub const EFI_SIMPLE_NETWORK_RECEIVE_PROMISCUOUS: u32 = 0x08;
+/// Receive packets in promiscuous multicast mode.
+pub const EFI_SIMPLE_NETWORK_RECEIVE_PROMISCUOUS_MULTICAST: u32 = 0x10;
```

Comment:
Use [`bitflags!`](https://docs.rs/bitflags/latest/bitflags/) here to define a new type, then you can use that new type instead of `u32` in the params to `receive_filters` and any other relevant places.